libraries/AP_HAL_ChibiOS/Util.cpp

```diff
@@ -169,24 +171,39 @@ bool Util::toneAlarm_init(uint8_t types)
 #endif
     _toneAlarm_types = types;
 
-#if !defined(HAL_PWM_ALARM) && !HAL_DSHOT_ALARM && !HAL_CANMANAGER_ENABLED
+#if !defined(HAL_PWM_ALARM) && !defined(HAL_PWM_ALT_ALARM) && !HAL_DSHOT_ALARM && !HAL_CANMANAGER_ENABLED
     // Nothing to do
     return false;
 #else
     return true;
 #endif
 }
 
+#if HAL_USE_PWM == TRUE
+bool Util::toneAlarm_init(const PWMConfig& pwm_cfg, PWMDriver* pwm_drv, pwmchannel_t chan, bool active_high)
+{
+    _toneAlarm_pwm_group.pwm_cfg = pwm_cfg;
+    _toneAlarm_pwm_group.pwm_drv = pwm_drv;
+    _toneAlarm_pwm_group.pwm_cfg.period = 1000;
```

Comment:
we might be able to use the existing clock, and scale the period, so we don't need to shutdown the other channels